docs/docs/specs/v0.2/graph.md

```diff
@@ -0,0 +1,122 @@
+# Graph API
+a proposal for `plugin.provides.graph`
+
+```tsx
+// let's say nodes are like this
+export type Node<T = any> = {
+  id: string;
+  label: string;
+  data: T;
+};
+
+const toNode = (e: any) => ({});
+
+// a trivial spaces plugin which renders documents as children to the space
+export const Plugin = () => {
+  return {
+    provides: {
+      graph: {
+        nodes: (parent: Node, publish: Function) => {
+          const space = parent.data;
+          if (!(space instanceof Object)) return; // instanceof SpaceProxy
+          const query = space.db.query();
+          const observer = query.observe((entity: any) => publish(toNode(entity)));
+          return () => observer.stop();
+        },
+      },
+    },
+  };
+};
+
+// what if we needed a static node under which to group some stuff 
+export const PluginNesting = () => {
+  return {
+    provides: {
+      graph: {
+        nodes: (parent: Node, upsert: Function, remove: Function) => {
+          const space = parent.data;
+          if (!(space instanceof Object)) return;
+          const query = space.db.query();
+          // make a static node
+          const group = upsert({
+            id: 'static_folder',
+            name: 'Documents',
+          });
+          const observer = query.observe({
+            added: (entity: any) => upsert(toNode(entity), group),
+            changed: (entity: any) => upsert(toNode(entity), group),
+            removed: (entity: any) => remove(toNode(entity), group),
+          });
+          return () => observer.stop();
+        },
+      },
+    },
+  };
+};
+
+// maybe it's cumbersome to pass a parent to the upsert/removes?
+// what if the upsert/remove functions were directly on the nodes?
+
+export type Node2<T = any> = {
+  id: string;
+  label: string;
+  data: T;
+  add(e: any): any;
+  remove(e: any): any;
+}
+
+export const PluginNesting2 = () => {
+  return {
+    provides: {
+      graph: {
+        nodes: (parent: Node2) => {
+          const space = parent.data;
```

Comment:
So, this is still hierarchical? Is there a way to get a node without referring to all of its parents, or traversing the tree to look for it?